qiita_db/metadata_template/constants.py

```diff
@@ -14,32 +14,32 @@
 # A dict containing the restrictions that apply to the sample templates
 SAMPLE_TEMPLATE_COLUMNS = {
     # The following columns are required by EBI for submission
-    'EBI': Restriction(columns={'collection_timestamp': 'timestamp',
-                                'physical_specimen_location': 'varchar',
-                                'taxon_id': 'integer',
-                                'scientific_name': 'varchar'},
+    'EBI': Restriction(columns={'collection_timestamp': 'datetime',
+                                'physical_specimen_location': 'str',
```

Comment:
Instead of storing the types as strings, use the actual types. This way, when checking the types in [this line](https://github.com/biocore/qiita/pull/1851/files#diff-fe8c1935d9373ee4e64757a20e79c142R1464) yo don't need to call `eval`, you will be able to do `datatype(val)`